data/spells/scripts/support/levitate.lua

```diff
@@ -0,0 +1,32 @@
+local function levitate(creature, parameter)
+	local fromPosition = creature:getPosition()
+
+	if parameter == "up" and fromPosition.z ~= 8 or parameter == "down" and fromPosition.z ~= 7 then
```

Comment:
Isn't it a trivial thing to do? i.e. I didn't even need to create the variable ``floorChange`` to hold the change of floor levels, but I chose to create it for clarity.

In the case your mentioning it does not add up for readability (if I understood you correctly.)